src/components/views/joinGame/JoinGame.tsx

```diff
@@ -0,0 +1,61 @@
+import BaseContainer from "../../ui/BaseContainer"
+import React, { useContext, useState } from "react"
+import Button from "@mui/material/Button"
+import SendIcon from "@mui/icons-material/Send"
+
+import { useGames } from "../../../hooks/api/useGames"
+import { Paths } from "../../routing/routers/Paths"
+import { useNavigate, useParams } from "react-router-dom"
+import { useApi } from "../../../hooks/api/useApi"
+import { appContext } from "../../../AppContext"
+
+const JoinGame = () => {
+  const { joinGame } = useGames()
+  const navigate = useNavigate()
+  const gameId = useParams()
+
+  const [loading, setLoading] = useState(false)
+  const { wrapApiCall, gameEntityController } = useApi()
+  const gameStore = useContext(appContext).playerStore
+
+  const findGame = async (gameId) => {
```

Comment:
i would just use the existing refreshGame funtion from useCurrentGame.
And your logic is similar to the GameGuard, except you handle everything here.

You need to think about the possible errors:

At start:
- you are already in the game
- The game may not exist
- The game is not in the state FINDING_PLAYERS
- (Later: ) The game is full
(And that may change while this component is displayed.

When you click the join button:
- you are already in the game (you opened it with 2 windows)
- The game may not exist
- The game is not in the state FINDING_PLAYERS
- (Later: ) The game is full
- Some other thing failed